### PR TITLE
[DFFE-11] Sign-Up Form Validation

### PR DESCRIPTION
**Pull request by OluwaboriShobayo**

### CODE REVIEW

**Comment by OluwaboriShobayo** [comment]
image
image
image

**Review by adaugochi — CHANGES_REQUESTED** [review]
`firstnameValidation, lastnameValidation and so on` function looks similar. Why don't you write a function and check if it is empty and another function if it matches regex.

**Comment by OluwaboriShobayo** [comment]
image
image
image


> `firstnameValidation, lastnameValidation and so on` function looks similar. Why don't you write a function and check if it is empty and another function if it matches regex.

Thank you, but when i tried creating different functions for them, it was not giving me the success criteria. The only criteria that all the fields were fulfilling was the 'Special numbers are not allowed"

File: submissions/oluwaboriShobayo/DOM/signUp.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+     numRegex = /^\(?([0-9]{3})\)?[-. ]?([0-9]{3} )[-. ]?([0-9]{4})$/;
+     emailRegex = /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/;
+     pwdRegex = /^(?=.*?[A-Z])(?=(.*[a-z]){1,})(?=(.*[\d]){1,})(?=(.*[\W]){1,})(?!.*\s).{8,}$/; 
+
+// // firstname VALIDATION
+function emptyField (event) {
+    if (firstname.value == '') {
+        firstname.nextElementSibling.innerHTML = "Please enter your firstname";
+        event.preventDefault();
+}
```
Comment (adaugochi) [inline submissions/oluwaboriShobayo/DOM/signUp.js:19]: You are still repeating one thing over and over again. Why don't you write a single function called `emptyField` that take `fieldName` and `errorMessage` as a parameter i.e `emptyField(fieldName, errorMessage, event)`. Then the function will hold this block of code as seen below 
```
if (fieldName.value == '') {
        fieldName.nextElementSibling.innerHTML = errorMessage;
        event.preventDefault();
}
```
Then when calling the function, you can then pass in the argument. E,g for firstname, you can call the function like this `emptyField(firstname, "Please enter your first name")`


Do a similar approach for the regex check

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Review by Ireti203 — APPROVED** [review]

**Pull request merged**